tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -44,6 +44,9 @@ public TraceContext(IDatadogTracer tracer, TraceTagCollection tags = null)
             Tags = tags ?? new TraceTagCollection(settings?.OutgoingTagPropagationHeaderMaxLength ?? TagPropagation.OutgoingTagPropagationHeaderMaxLength);
         }
 
+        // Gets the spans. For testing only.
+        internal ArrayBuilder<Span> Spans => _spans;
```

Comment:
This seems more useful than `AddTaintedForTest` but also a little more likely to be misused elsewhere. I don't have a solution, but is there a better way of catching the span count? Could `TraceContext` be mocked in someway?

Reply:
I could also try with reflection... I will check.

Reply:
Same as above, reflection is a solution that works for this case.